tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -53,127 +58,34 @@ public void run() {
-    }
-
-    private boolean isLoginUser(HttpRequest request) {
-        return request.getSessionId()
-                .map(this::isLoginUser)
-                .orElse(false);
-    }
-
-    private boolean isLoginUser(String sessionId) {
-        Session session = SESSION_MANAGER.findSession(sessionId);
-        if (session == null) {
-            return false;
-        }
-        return session.contains("user");
-    }
-
-    private HttpResponse getFileResponse(String requestPath) {
         try {
-            URL resource = getClass().getClassLoader().getResource("static/" + requestPath);
-            if (resource == null) {
-                throw new IllegalArgumentException("존재하지 않는 자원입니다: " + requestPath);
-            }
-            ResponseFile responseFile = ResponseFile.of(resource);
-            return HttpResponse.createFileResponse(responseFile);
-        } catch (Exception e) {
-            log.error(e.getMessage(), e);
-            return HttpResponse.createRedirectResponse(
-                    HttpStatus.FOUND,
-                    "/404.html"
-            );
-        }
-    }
-
-    private HttpResponse register(HttpRequest request) {
-        Queries queries = Queries.of(request.getBody());
-        String account = queries.get("account");
+            HttpResponse response = new HttpResponse();
+            Controller controller = requestMapping.getController(request);
+            controller.service(request, response);
 
-        User user = new User(account, queries.get("password"), queries.get("email"));
-
-        if (InMemoryUserRepository.findByAccount(account).isPresent()) {
-            String message = "이미 존재하는 사용자입니다: " + account;
-            log.warn(message);
-            return HttpResponse.createTextResponse(HttpStatus.CONFLICT, message);
-        }
-        InMemoryUserRepository.save(user);
-        return HttpResponse.createRedirectResponse(HttpStatus.FOUND, "/index.html");
-    }
-
-    private HttpResponse login(HttpRequest request) {
-        String requestBody = request.getBody();
-        Queries queries = Queries.of(requestBody);
-        String account = queries.get("account");
-        String password = queries.get("password");
-        validateLoginRequest(account, password);
-
-        try {
-            User user = getLoginUser(account, password);
-            Session session = sessionGenerator.create();
-            session.setAttribute("user", user);
-            SESSION_MANAGER.add(session);
-            ResponseCookie sessionCookie = ResponseCookie.of(session);
-            Map<String, String> headers = new HashMap<>();
-            headers.put("Location", "/index.html");
-            headers.put("Set-Cookie", sessionCookie.toResponse());
-            return new HttpResponse(HttpStatus.FOUND, headers);
+            return response;
+        } catch (FileException fileException) {
+            return HttpResponse.createRedirectResponse(HttpStatus.FOUND, "/404.html");
```

Comment:
Response는 이미 타입으로 표현하고 있으니 중복해서 표시할 필요는 없겠네요.

```suggestion
            return HttpResponse.orRedirect("/404.html");
```